### PR TITLE
Demo viz

### PR DESCRIPTION
**Pull request by alexsax**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] I ran `tests/hw_manual_test.py` on hardware and observed that robot behavior looks normal.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -25,3 +25,6 @@
 [submodule "src/home_robot/home_robot/perception/detection/grounded_sam/Grounded-Segment-Anything"]
 	path = src/home_robot/home_robot/perception/detection/grounded_sam/Grounded-Segment-Anything
 	url = https://github.com/IDEA-Research/Grounded-Segment-Anything
+[submodule "src/home_robot/home_robot/perception/minigpt4/MiniGPT-4"]
```
Comment (cpaxton) [inline .gitmodules:28]: Do we want to add this as a submodule?

File: projects/scannet_offline_eval/configs/model/instancemap3d_top_down.yaml [hunk 2]
```diff
@@ -3,6 +3,6 @@ defaults:
   - semantic_sensor: grounded_sam_perception
   - voxel_map: default
 
-_target_: build_sparse_voxel_map.SparseVoxelMapAgent
+_target_: home_robot.agent.multitask.sparse_voxel_instance_map.SparseVoxelMapAgent
```
Comment (cpaxton) [inline projects/scannet_offline_eval/configs/model/instancemap3d_top_down.yaml:6]: what is this file doing?

File: projects/scannet_offline_eval/create_instance_map.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    Args:
+        model (SparseVoxelMapAgent): _description_
+        dataset (ScanNetDataset): _description_
+        scene_name (str, optional): _description_. Defaults to 'scene0192_00'.
+        show_backend (Optional[str], optional): _description_. Defaults to 'open3d'.
+        torch_device (str, optional): _description_. Defaults to 'cuda'.
+    """
+    print(dataset.scene_list)
+    idx = dataset.scene_list.index(scene_name)
+    scene_obs = dataset.__getitem__(idx, show_progress=True)
```
Comment (cpaxton) [inline projects/scannet_offline_eval/create_instance_map.py:43]: why not use square brackets here

File: projects/scannet_offline_eval/demo/proto_robot/mock_publisher.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import pickle
+
+import grpc
+import robodata_pb2
+import robodata_pb2_grpc
+import torch
+from PIL import Image
+from torchvision import transforms
+
+stretch_filename = "/private/home/ssax/home-robot/src/home_robot/home_robot/datasets/robot/data/stretch_/2023-09-06-demo01.pkl"
```
Comment (cpaxton) [inline projects/scannet_offline_eval/demo/proto_robot/mock_publisher.py:12]: hard-coded filename, we should remove this right

File: projects/scannet_offline_eval/demo/proto_robot/mock_publisher.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+data = pickle.load(open(stretch_filename, "rb"))
+
+img_filename = "proto_robot/dog.webp"
```
Comment (cpaxton) [inline projects/scannet_offline_eval/demo/proto_robot/mock_publisher.py:15]: so many webps in the github repo

File: projects/scannet_offline_eval/demo/components/app.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+@dataclass
+class AppConfig:
+    pointcloud_update_freq_ms: int = 2000
+    video_feed_update_freq_ms: int = 300
+
+    directory_watcher_update_freq_ms: int = 300
+    directory_watch_path: str = "publishers/published_trajectory/obs"
+    pointcloud_voxel_size: float = 0.05
+
```
Comment (cpaxton) [inline projects/scannet_offline_eval/demo/components/app.py:36]: I think a lot of these files make more sense in home_robot/datasets or something right?

**Review by cpaxton — COMMENTED** [review]
So far looks good, i think we should not add so many images, avoid hard coded paths (relative), and maybe we can move these components under the core home_robot repo?

File: .gitmodules (hunk 1 above)
Comment (alexsax) [reply .gitmodules:28]: Oof I have no idea where this came from -- I'll remove it from the commit history

File: projects/scannet_offline_eval/configs/model/instancemap3d_top_down.yaml (hunk 2 above)
Comment (alexsax) [reply projects/scannet_offline_eval/configs/model/instancemap3d_top_down.yaml:6]: This implements `build_representation_and_return_queries` -- it's the "final form" that consumes posed RGBD frames and text, and returns a response. The style that the rest of the benchmark tasks will follow

File: .gitmodules (hunk 1 above)
Comment (cpaxton) [reply .gitmodules:28]: thanks. it is in the demo grasp branch but i'm not sure we want to keep it

File: projects/scannet_offline_eval/demo/assets/images/googley_roomba.jpeg
Comment (cpaxton) [inline projects/scannet_offline_eval/demo/assets/images/googley_roomba.jpeg:1]: I'm a bit nervous about adding larger binary (lfs) files to our git repo, wont that make things slow?
i guess with lfs this is fine

File: projects/scannet_offline_eval/demo/assets/images/spot_small.jpeg
Comment (cpaxton) [inline projects/scannet_offline_eval/demo/assets/images/spot_small.jpeg:1]: why are we adding this one

File: projects/scannet_offline_eval/demo/assets/images/stream_paused_4_3.jpg
Comment (cpaxton) [inline projects/scannet_offline_eval/demo/assets/images/stream_paused_4_3.jpg:1]: Where are these images  from? should we add sourcing info / credit to the readme?

File: projects/scannet_offline_eval/demo/assets/prompts/demo.txt [hunk 7]
```diff
@@ -0,0 +1,25 @@
+Pretend that you are a professional Guide who can direct unsighted people and robots in order to accomomplish tasks.
```
Comment (cpaxton) [inline projects/scannet_offline_eval/demo/assets/prompts/demo.txt:1]: What is this for?
Should we link this in the demo.md?

**Comment by cpaxton** [comment]
merged to demo branch

**Pull request closed without merging**